tests/integration/metrics_test.go

```diff
@@ -228,3 +234,167 @@ func TestExitedContainersWithoutTTL(t *testing.T) {
+
+	// Create our mocked container cgroups filesystem in the tempDir directory
+	// that will be a symLink to our cgroups testdata
+	cgroupsFolder := filepath.Join(hostRoot, "cgroup")
+	mockFilesystem, err := filepath.Abs(filepath.Join(relativePathToTestdataFileystem, "cgroup"))
+	require.NoError(t, err)
+
+	err = os.Symlink(mockFilesystem, cgroupsFolder)
+	require.NoError(t, err)
+
+	// Create mocked directory proc
+	err = os.Mkdir(filepath.Join(hostRoot, "proc"), 0755)
+	require.NoError(t, err)
+
+	err = generateMockedProcMountsFile(cgroupsFolder, hostRoot)
+	require.NoError(t, err)
+
+	err = generateMockedProcPIDCGroupFile(hostRoot)
+	require.NoError(t, err)
+
+	err = genereateMockedProcNetDevFile(hostRoot)
+	require.NoError(t, err)
+
+	// CgroupsFetcherMock is the raw CgroupsFetcher with mocked cpu.systemUsage and time
+	// The hostRoot is our mocked filesystem
+	cgroupFetcher, err := NewCgroupsFetcherMock(hostRoot, "cgroupfs", "", mockedTimeForAllMetricsTest, 19026130000000)
+	require.NoError(t, err)
+
+	storer := generateInMemoryStorerWithPreviousCPUState()
+	inspector := NewInspectorMock(InspectorContainerID, InspectorPID, 2)
+
+	t.Run("Given a mockedFilesystem and previous CPU state Then processed metrics are as expected", func(t *testing.T) {
+		metrics := biz.NewProcessor(storer, cgroupFetcher, inspector, 0)
+
+		sample, err := metrics.Process(InspectorContainerID)
+		require.NoError(t, err)
+		assert.Equal(t, expectedSample, sample)
+	})
+}
+
+// generateInMemoryStorerWithPreviousCPUState creates a storere with a previous CPU state
+// in order to make the processor calculate a Cpu Delta
+func generateInMemoryStorerWithPreviousCPUState() persist.Storer {
+	var previous struct {
+		Time int64
+		CPU  raw.CPU
+	}
+
+	storer := persist.NewInMemoryStore()
+	// We set the time as 10 seconds before the timestamp for the metrics
+	previous.Time = mockedTimeForAllMetricsTest.Add(-time.Second * 10).Unix()
+	previous.CPU = raw.CPU{
+		TotalUsage:        1,
+		UsageInUsermode:   1,
+		UsageInKernelmode: 1,
+		PercpuUsage:       nil,
+		ThrottledPeriods:  1,
+		ThrottledTimeNS:   1,
+		SystemUsage:       1,
+		OnlineCPUs:        1,
+		Shares:            1,
+	}
+	storer.Set(InspectorContainerID, previous)
+	return storer
+}
+
+func generateMockedProcMountsFile(cgroupsFolder, hostRoot string) error {
+	mountsContent := `cgroup <HOST_ROOT>/blkio cgroup rw,nosuid,nodev,noexec,relatime,blkio 0 0
+cgroup <HOST_ROOT>/pids cgroup rw,nosuid,nodev,noexec,relatime,pids 0 0
+cgroup <HOST_ROOT>/cpu,cpuacct cgroup rw,nosuid,nodev,noexec,relatime,cpu,cpuacct 0 0
+cgroup <HOST_ROOT>/cpuset cgroup rw,nosuid,nodev,noexec,relatime,cpuset 0 0
+cgroup <HOST_ROOT>/memory cgroup rw,nosuid,nodev,noexec,relatime,memory 0 0
+`
+	mountsContent = strings.ReplaceAll(mountsContent, "<HOST_ROOT>", cgroupsFolder)
+	return os.WriteFile(filepath.Join(hostRoot, "proc", "mounts"), []byte(mountsContent), 0755)
+}
+
+func generateMockedProcPIDCGroupFile(hostRoot string) error {
```

Comment:
couldn't this and the NetDevFile be with the others files in `/testdata` . I guess we should need to 'fix' the InspectorPID but could simplify the code a lot i think.

Reply:
They are in the testdata, but here they are just copied to the specific folder where the code is looking at but done in the hostRoot that is the tempDir.

Reply:
so yes, this files are already in the testdata.